src/hooks/useDisableScrolling.ts

```diff
@@ -0,0 +1,12 @@
+import { useEffect } from 'react'
+import { isMobile } from 'utils/userAgent'
+
+export default function useDisableScrolling(disable: boolean | undefined | null) {
+  useEffect(() => {
+    if (disable && isMobile) {
+      document.body.style.overflow = 'hidden'
+    } else if (isMobile) {
+      document.body.style.overflow = 'auto'
+    }
```

Comment:
Simplify the mobile check by making the whole thing a nop:

```suggestion
    if (!isMobile) return
    document.body.style.overflow = disable ? 'hidden' : 'auto'
```